controller/scan/harbor.go

```diff
@@ -0,0 +1,191 @@
+package scan
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/neuvector/neuvector/share"
+	"github.com/neuvector/neuvector/share/httpclient"
+	log "github.com/sirupsen/logrus"
+)
+
+type harbor struct {
+	base
+	client http.Client
+}
+
+type HarborApiProject struct {
+	Id         int     `json:"id"`
+	RegistryId *string `json:"registry_id"`
+}
+
+func (proj *HarborApiProject) IsProxyCacheProject() bool {
+	return proj.RegistryId != nil
+}
+
+type HarborApiRepository struct {
+	FullName  string `json:"name"`
+	ProjectId int    `json:"project_id"`
+}
+
+func newHarbor(baseDriver base) *harbor {
+	client := &http.Client{
+		Timeout: gitTimeout,
+	}
+
+	t, err := httpclient.GetTransport(baseDriver.proxy)
+	if err != nil {
+		log.WithError(err).Warn("failed to get transport")
+	} else {
+		client.Transport = t
+	}
+
+	return &harbor{
+		base:   baseDriver,
+		client: *client,
+	}
+}
+
+// TODO: the splits in projectName() and repoName() are duplicated, perform only once
+func (harborRepo *HarborApiRepository) projectName() string {
```

Comment:
It would be nice to have a unit test for this function, so folks know what is a valid project name and what is not without having to install harbor.